benchmark/framework/aea_test_wrapper.py

```diff
@@ -0,0 +1,209 @@
+
+
+class AEATestWrapper:
+    """A testing wrapper to run and control an agent."""
+
+    def __init__(self, name: str = "my_aea", skills: List[dict] = None):
+        """Make an agency with optional name and skills."""
+        self.skills = skills or []
+        self.name = name
+
+        self.aea = self.make_aea(
+            self.name, [self.make_skill(**skill) for skill in self.skills]
+        )
+
+    def make_aea(self, name: str = "my_eae", components: List[Component] = None) -> AEA:
+        """
+        Create AEA from name and already loaded components.
+
+        :return: AEA
+        """
+        components = components or []
+        builder = AEABuilder()
+
+        builder.set_name(self.name)
+        builder.add_private_key(FETCHAI, "")
+
+        for component in components:
+            builder._resources.add_component(component)
+
+        aea = builder.build()
+        return aea
+
+    def make_skill(
+        self,
+        config: SkillConfig = None,
+        context: SkillContext = None,
+        handlers: Optional[Dict[str, Type[Handler]]] = None,
+    ) -> Skill:
+        """
+        Make a skill from optional config, context, handlers dict.
+
+        :return: Skill
+        """
+        handlers = handlers or {}
+        context = context or SkillContext()
+        config = config or SkillConfig()
+
+        handlers_instances = {
+            name: cls(name=name, skill_context=context)
+            for name, cls in handlers.items()
+        }
+
+        skill = Skill(
+            configuration=config, skill_context=context, handlers=handlers_instances
+        )
+
+        return skill
+
+    @classmethod
+    def dummy_default_message(
+        cls,
+        dialogue_reference: Tuple[str, str] = ("", ""),
+        message_id: int = 1,
+        target: int = 0,
+        performative: DefaultMessage.Performative = DefaultMessage.Performative.BYTES,
+        content: Union[str, bytes] = "hello world!",
+    ) -> Message:
+        """
+        Construct simple message, all arguments are optional.
+
+        :return: Message
+        """
+        if isinstance(content, str):
+            content = content.encode("utf-8")
+
+        return DefaultMessage(
+            dialogue_reference=dialogue_reference,
+            message_id=message_id,
+            target=target,
+            performative=performative,
+            content=content,
+        )
+
+    @classmethod
+    def dummy_envelope(
+        cls,
+        to: str = "test",
+        sender: str = "test",
+        protocol_id: ProtocolId = DefaultMessage.protocol_id,
+        message: Message = None,
+    ) -> Envelope:
+        """
+        Create envelope, if message is not passed use .dummy_message method.
+
+        :return: Envelope
+        """
+        message = message or cls.dummy_default_message()
+        return Envelope(
+            to=to,
+            sender=sender,
+            protocol_id=protocol_id,
```

Comment:
hard code `protocol_id = DefaultMessage.protocol_id`

the reason, you are hardcoding the serializer as well, and they need to match!

Reply:
done